tensorboard/webapp/feature_flag/store/feature_flag_selectors.ts

```diff
@@ -58,6 +58,24 @@ export const getFeatureFlagsMetadata = createSelector(
   }
 );
 
+export const getFeatureFlagsToSendToServer = createSelector(
+  selectFeatureFlagState,
+  (state: FeatureFlagState): Partial<FeatureFlags> => {
+    let featureFlagsToSendToServer: Partial<FeatureFlags> = {};
+    for (const entry in state.flagOverrides) {
+      const entryMetadata = state.metadata[entry as keyof FeatureFlags];
+      if (!entryMetadata.queryParamOverride || !entryMetadata.sendToServer) {
+        continue;
+      }
+      featureFlagsToSendToServer = {
+        ...featureFlagsToSendToServer,
+        [entry]: state.flagOverrides[entry as keyof FeatureFlags],
+      };
+    }
+    return featureFlagsToSendToServer;
```

Comment:
> Unfortunately not totally that simple. Making that change leads to the following error:
> 
> Type 'number | boolean | string[] | null | undefined' is not assignable to type 'never'.
> 
> The anti-pattern you identifeid here was a workaround. But I agree from a runtime complexity standpoint its pretty ugly. So I've changed to the other workaround: type featureFlagsToSendToServer as any, initially, and then cast to Partial when returning it. How do you feel about that?

You can get away with it with some slightly looser typing.
```suggestion
    let featureFlagsToSendToServer: Partial<Record<keyof FeatureFlags, FeatureFlagType>> = {};
    for (const entry in state.flagOverrides) {
      const entryMetadata = state.metadata[entry as keyof FeatureFlags];
      if (
        entryMetadata.queryParamOverride &&
        entryMetadata.sendToServerWhenOverridden
      ) {
        featureFlagsToSendToServer[entry as keyof FeatureFlags] = state.flagOverrides[entry as keyof FeatureFlags];
      }
    }
    return featureFlagsToSendToServer as Partial<FeatureFlags>;
```

Reply:
Done.